### PR TITLE
Getting ready for releasing Azure.Core

### PR DESCRIPTION
**Pull request by maririos**

### CODE REVIEW

**Review by tg-msft — APPROVED** [review]

**Comment by AlexGhiondea** [comment]
It looks like the failures are the expected ones (can't restore -preview.6 as that is not yet released) -- so that's a good sign that we are using package references to Azure.Core.

File: eng/Directory.Build.Data.props [hunk 1]
```diff
@@ -84,7 +84,7 @@
     <DelaySign>false</DelaySign>
     <PublicSign>false</PublicSign>
     <ImportDefaultReferences>false</ImportDefaultReferences>
-    <UseProjectReferenceToAzureCore>true</UseProjectReferenceToAzureCore>
+    <UseProjectReferenceToAzureCore>false</UseProjectReferenceToAzureCore>
```
Comment (jsquire) [inline eng/Directory.Build.Data.props]: Nice catch!

Seems like this should be the default state at the `eng` level, with individual libraries overriding if they need the latest and greatest.

**Review by jsquire — APPROVED** [review]

**Comment by maririos** [comment]
Note that I couldn't trigger the CI pipelines `/azp run net - keyvault - ci` so I had to add `\n` to the readme to kick the `net - core ci`

File: eng/Directory.Build.Data.props (hunk 1 above)
Comment (maririos) [reply eng/Directory.Build.Data.props]: My understanding is that we wanted it to be `true` while we were all adopting the changes that were coming to Core. 
Now that we are getting ready to ship, each client library will be able to override, like you say

**Review by KrzysztofCwalina — APPROVED** [review]

**Comment by maririos** [comment]
@jsquire net - servicebus - tests failed . Do you have context on why? From the error, it is not related to Core, but want to double check with someone that has worked on that library

**Comment by jsquire** [comment]
> @jsquire `net - servicebus - tests` failed . Do you have context on why? From the error, it is not related to Core, but want to double check with someone that has worked on that library

Looks like one of the transient failures at a glance;  there hasn't been any work in SB for track two, so it has no awareness of Core.    There's not been anything outside of RBAC work that has been done to track one as far as I'm aware, and the scope wouldn't have impact here.

**Comment by jsquire** [comment]
@maririos:  I see the Event Hubs test are running.   Expect failures for track one; the tests are notoriously flaky since migration.   We haven't had time to fully revisit and stabilize.   I'll loop back around and verify that the track two test were good; that's got a Core dependency, but just on the `TokenCredential` type definition and no functionality.

**Comment by maririos** [comment]
`net - keyvault - tests ` is failing but with a known Track1 issue. track2 test are passing so we are good to go in terms of KeyVault

**Comment by jsquire** [comment]
> @maririos: I see the Event Hubs test are running. Expect failures for track one; the tests are notoriously flaky since migration. We haven't had time to fully revisit and stabilize. I'll loop back around and verify that the track two test were good; that's got a Core dependency, but just on the `TokenCredential` type definition and no functionality.

I should also mention that the EH tests for Track One will take ~5.5 hours to complete.    I'm confident in the current state, given our very tiny reliance on core.   I've looked at the source and can confirm the signature matches our expected.   I don't see a need to wait for completion.

**Comment by maririos** [comment]
Thanks @jsquire , then let's merge this thing!!!

File: eng/Directory.Build.Data.props (hunk 1 above)
Comment (weshaggard) [reply eng/Directory.Build.Data.props]: There was some discussion in the past about setting this to true in the Azure.Core.props file when you are building in VS.  That has potential issues if things get out of sync but while things are in sync we can use a package reference for our command line/official builds and use a project reference for local devs that are working in VS.

**Pull request merged**